### ISSUE TITLE
How do I call an API for verified user to post images only?

### ISSUE THREAD

**Issue by ghost** [buaazp/zimg#34]
1. How do I call an API for only verified user to post images?
   e.g. server response code of "1" from another server with a submitted string and then only allow for upload.
2. and how can i change the image naming? instead of md5ing it, i would like it to have a submitted query as filename (can u point to the file location for modification for this?)

**Comment by Niklas81** [comment]
For me, authentication is best done with php before upload. However, it would be really helpful if filename could be set while uploading using curl. Using only an MD5 hash that I as a programmer has no insight into is bad practice, and would lock me into zimg. A much better soulution would be if I can use e.g. this to upload a file and save as myfilname:

curl -F "blob=@testup.jpeg;type=image/jpeg" "http://127.0.0.1:4869/upload?filename=myfilname"

above command would then save the file as myfilname.jpg

Also, I have a concern regarding prevention of DDoS / overload-attacks. Since Zimg is loaded directly, I cannot do normal checks using nginx or php against e.g. ip address or session. So it would be really helpful if Zimg would have some features to handle this.

So, in summary:

Question 1: Any chance you could implement this naming-feature? 
Question 2: Any way of preventing  DDoS / overload-attacks? If not, any plans for implementing this?
Question 3: In terms of long-term maintainability, are you planning to maintain zimg for many years to come?

Best regards

**Comment by Niklas81** [comment]
Oh, and to download file by stating filename instead of md5, one could use perhaps:
http://127.0.0.1:4869/?filename=myfilename

without file extension (so different formats can be downloaded)

**Comment by Niklas81** [comment]
One more thing. In this benchmark comparison:

http://zimg.buaa.us/documents/benchmark_v1/

Did you not use memcached when running it under Nginx+PHP, whilst comparing it to zimg where you did use memcached? If so, I'd like to see how Nginx+PHP+memcached performs against zimg+memcaced or how Nginx+PHP performs against just zimg.. :)

**Comment by buaazp** [comment]
@Niklas81 Thanks for you suggestions.

1. The naming-feature depends on a map between MD5 and expect-name. I should use some way to store the map when POST images and query it before GET images which costs high latency. In fact, if you use zimg, you should store some related information like user name, password and any other data in the database. You can store the map of filename by the way. So, zimg will not support this feature.
2. There is a memcached as cache in front of the hdd storage.
3. Now zimg has no solution of preventing attacks. You can use zimg behind a nginx to achieve this target.
4. I will maintain zimg until the new version of my image storage and processing system release. Which have most of features of zimg and is in developing.

**Comment by buaazp** [comment]
@qswl 
using php-curl to post image to zimg: #45 and #97

**Comment by r3wt** [comment]
@buaazp I like zimg so much, i release an opensource library for it for php :-) :+1:  soon
